source/val/validate_decorations.cpp

```diff
@@ -909,6 +909,13 @@ spv_result_t CheckDecorationsOfBuffers(ValidationState_t& vstate) {
                     : (push_constant ? "PushConstant" : "StorageBuffer");
 
         if (spvIsVulkanEnv(vstate.context()->target_env)) {
+
+          if(storage_buffer && hasDecoration(id, SpvDecorationBlock, vstate)){
```

Comment:
In debug mode, we never get to this line. Why?

Reply:
If you check for SpvDecorationBufferBlock here, the test fails as expected. Not sure what you mean by not getting here.